include/ofi_hmem.h

```diff
@@ -127,6 +127,7 @@ struct ofi_hmem_ops {
 			     size_t *size);
 	bool (*is_ipc_enabled)(void);
 	int (*get_ipc_handle_size)(size_t *size);
+	int (*get_async_copy_cutoff)(size_t *size);
```

Comment:
If this is an environment variable, why add it to the hmem ops to look it up? This is a lot of code added to all of the other interfaces just to add a param_get inside the rocr code.

Reply:
The primary reason I did this is because I wanted to call it from SHM provider. The SHM provider makes a decision based on this value and the size of the message it's progressing to either do an asynchronous memory op or synchronous. When I was testing, I found that synchronous calls out perform asynchronous calls for message sizes below 512 KB. If the asynchronous approach is adopted by other device types, this same performance aspect might apply to them. In this case the cutoff value might differ between different devices, and it would be more generic to use the approach in this patch.

Reply:
Until we have another implementation, it's not worth carrying the extra complexity in the hope that it may someday be useful.  It's too difficult to predict what might be needed.  The cutoff may be more closely related to the actual device, not what API it's accessed through.

Reply:
I removed the patch which adds this for now. I'll create a separate PR for this particular improvement.